recipes/dlib/all/conanfile.py

```diff
@@ -90,107 +83,107 @@ def requirements(self):
-        cmake = CMake(self)
+    def generate(self):
+        tc = CMakeToolchain(self)
 
         # With in-project builds dlib is always built as a static library,
         # we want to be able to build it as a shared library too
-        cmake.definitions["DLIB_IN_PROJECT_BUILD"] = False
-
-        cmake.definitions["DLIB_ISO_CPP_ONLY"] = False
-        cmake.definitions["DLIB_NO_GUI_SUPPORT"] = True
+        tc.variables["DLIB_IN_PROJECT_BUILD"] = False
 
+        tc.variables["DLIB_ISO_CPP_ONLY"] = False
+        tc.variables["DLIB_NO_GUI_SUPPORT"] = True
         # Configure external dependencies
-        cmake.definitions["DLIB_JPEG_SUPPORT"] = self.options.with_jpeg
+        tc.variables["DLIB_JPEG_SUPPORT"] = self.options.with_jpeg
         if self._has_with_webp_option:
-            cmake.definitions["DLIB_WEBP_SUPPORT"] = self.options.with_webp
-        cmake.definitions["DLIB_LINK_WITH_SQLITE3"] = self.options.with_sqlite3
-        cmake.definitions["DLIB_USE_BLAS"] = True    # FIXME: all the logic behind is not sufficiently under control
-        cmake.definitions["DLIB_USE_LAPACK"] = True  # FIXME: all the logic behind is not sufficiently under control
-        cmake.definitions["DLIB_USE_CUDA"] = False   # TODO: add with_cuda option?
-        cmake.definitions["DLIB_PNG_SUPPORT"] = self.options.with_png
-        cmake.definitions["DLIB_GIF_SUPPORT"] = self.options.with_gif
-        cmake.definitions["DLIB_USE_MKL_FFT"] = False
+            tc.variables["DLIB_WEBP_SUPPORT"] = self.options.with_webp
+        tc.variables["DLIB_LINK_WITH_SQLITE3"] = self.options.with_sqlite3
+        tc.variables["DLIB_USE_BLAS"] = True    # FIXME: all the logic behind is not sufficiently under control
+        tc.variables["DLIB_USE_LAPACK"] = True  # FIXME: all the logic behind is not sufficiently under control
+        tc.variables["DLIB_USE_CUDA"] = False   # TODO: add with_cuda option?
+        tc.variables["DLIB_PNG_SUPPORT"] = self.options.with_png
+        tc.variables["DLIB_GIF_SUPPORT"] = self.options.with_gif
+        tc.variables["DLIB_USE_MKL_FFT"] = False
 
         # Configure SIMD options if possible
         if self.settings.arch in ["x86", "x86_64"]:
             if self.options.with_sse2 != "auto":
-                cmake.definitions["USE_SSE2_INSTRUCTIONS"] = self.options.with_sse2
+                tc.variables["USE_SSE2_INSTRUCTIONS"] = self.options.with_sse2
             if self.options.with_sse4 != "auto":
-                cmake.definitions["USE_SSE4_INSTRUCTIONS"] = self.options.with_sse4
+                tc.variables["USE_SSE4_INSTRUCTIONS"] = self.options.with_sse4
             if self.options.with_avx != "auto":
-                cmake.definitions["USE_AVX_INSTRUCTIONS"] = self.options.with_avx
-
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+                tc.variables["USE_AVX_INSTRUCTIONS"] = self.options.with_avx
+        tc.generate()
+        tc = CMakeDeps(self)
```

Comment:
Is it a good idea to reuse `tc` instead of defining a new variable.
The migration guides suggest:
```
tc = CMakeToolchain(self)
...
deps = CMakeDeps(self)
```

I have not looked at other recipes so can't comment on the consistency in CCI.

Reply:
Indeed is a code smell! You will find it widespread in CCI, including the [package templates](https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates). But we only use it in `generate()` scope, and as it's small, so there is no big deal about doing a mess with the same variable.
However, you can use other variables indeed, or even, call directly without variable `CMakeDeps(self).generate()`.